infra/Pos/Communication/Relay/Logic.hs

```diff
@@ -120,6 +121,34 @@ handleReqL proxy = reifyMsgLimit (Proxy @(ReqMsg key tag)) $
     constructDataMsg :: contents -> InvOrData tag key contents
     constructDataMsg = Right . DataMsg
 
+handleMempoolL
+    :: forall key tag contents m .
+       ( Bi (MempoolMsg tag)
+       , Bi (InvOrData tag key contents)
+       , Message (InvOrData tag key contents)
+       , Relay m tag key contents
+       , MinRelayWorkMode m
+       , MonadDBLimits m
+       )
+    => RelayProxy key tag contents
+    -> m (ListenerSpec m, OutSpecs)
+handleMempoolL proxy = reifyMsgLimit (Proxy @(MempoolMsg tag)) $
+  \(_ :: Proxy s) -> return $ listenerConv $
+    \_ __peerId ConversationActions{..} ->
+    whenJustM recv $ \(withLimitedLength @s -> msg@MempoolMsg{..}) -> do
```

Comment:
I think we should spend some time to make this API more usable, convenient and nice :confused: 
It's not problems of this commit, just random thought about `time-warp-nt`.